src/mongocxx/test/search_index_view.cpp

```diff
@@ -1 +1,211 @@
+        auto model1 = search_index_model(name1, definition1.view());
+
+        // {
+        //   name: 'test-search-index-2',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name2 = "test-search-index-2";
+        auto definition2 = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model2 = search_index_model(name2, definition2.view());
+
+        std::vector<search_index_model> models = {model1, model2};
+
+        std::vector<std::string> result = siv.create_many(models);
+        std::vector<std::string> expected = {"test-search-index-1", "test-search-index-2"};
+        REQUIRE(result == expected);
+
+        wait_for_search_index(siv, model1, true, false);
+        wait_for_search_index(siv, model2, true, false);
+
+        std::cout << "create many SUCCESS" << std::endl;
+    }
+
+    SECTION("drop one") {
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
+        auto siv = coll.search_indexes();
+        // {
+        //   name: 'test-search-index',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name = "test-search-index";
+        auto definition = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model = search_index_model(name, definition.view());
+
+        REQUIRE(siv.create_one(model) == "test-search-index");
+
+        wait_for_search_index(siv, model, true, false);
+
+        siv.drop_one(name);
+
+        wait_for_search_index(siv, model, false, false);
+
+        std::cout << "drop one SUCCESS" << std::endl;
+    }
+
+    SECTION("update one") {
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
+        auto siv = coll.search_indexes();
+        // {
+        //   name: 'test-search-index',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name = "test-search-index";
+        auto definition = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model = search_index_model(name, definition.view());
+
+        REQUIRE(siv.create_one(model) == "test-search-index");
+
+        wait_for_search_index(siv, model, true, false);
+
+        // definition : {
+        //   mappings : { dynamic: true }
+        // }
+        auto new_definition = make_document(kvp("mappings", make_document(kvp("dynamic", true))));
+        auto new_model = search_index_model(name, new_definition.view());
+        siv.update_one(name, new_definition.view());
+
+        wait_for_search_index(siv, new_model, true, true);
+
+        std::cout << "update one SUCCESS" << std::endl;
+    }
+
+    // SECTION("drop one suppress namespace not found") {
```

Comment:
This test case comes from https://github.com/mongodb/specifications/commit/267a54d1bb9af894d05d3dd1a6bda0fa53e20a3f#diff-7f618607e8afab57c7ebfd80f0e91802e2f705ec79bfb946947d026a4fa97ba0R189. I'm not sure how to create a driver-side collection object in the CXX. I see this private function in `collection.hpp` that seems to be the best option at this point.
```
    MONGOCXX_PRIVATE collection(const database& database,
                                bsoncxx::string::view_or_value collection_name);
```
Currently, the commented out test case returns this
```
invalid use of default constructed or moved-from mongocxx::collection object
```
due to there not being a valid `_impl` initialized with the collection object.

Reply:
I don't believe that this works either. I get this error returned: 
```
  Collection 'test.64cbcd93b74f63a863033401' does not exist.: generic server
  error
```
It's failing on the `search_indexes` call, not the `drop_one` call.

Reply:
> It's failing on the `search_indexes` call, not the `drop_one` call.

I think it is failing on the `drop_one` call.

The specification notes a requirement to ignore the "NamespaceNotFound" error.

> Drivers MUST suppress NamespaceNotFound errors for the `dropSearchIndex` helper.

I expect `drop_one` needs to be updated. Try applying this commit: https://github.com/mongodb/mongo-cxx-driver/commit/23c8bd33014acc9a1107b0d05e351cbf819b95a2

Reply:
Got it, thanks. Fixed now.